jps-plugin/src/org/intellij/erlang/jps/builder/ErlangBuilder.java

```diff
@@ -446,84 +406,23 @@ private static void consumeFiles(@NotNull BuildOutputConsumer outputConsumer,
-      ContainerUtil.addIfNotNull(myDirtyElements, getDirtyElement(erlangTarget, file, erlangSourceRootDescriptor));
-      return true;
-    }
-
-    @Nullable
-    protected abstract T getDirtyElement(@NotNull ErlangTarget target,
-                                         @NotNull File file,
-                                         @NotNull ErlangSourceRootDescriptor root) throws IOException;
-  }
-
-  private static class Node {
-    final String myErlangModulePath;
-    final List<Node> myDependencies = ContainerUtil.newArrayList();
-    public boolean myDirty = false;
-
-    Node(String nodeName) {
-      myErlangModulePath = nodeName;
-    }
-  }
-
-  private static class SortedModuleDependencyGraph implements GraphGenerator.SemiGraph<Node> {
-    private final LinkedHashMap<String, Node> myNodePathsMap;
-
-    public SortedModuleDependencyGraph(List<ErlangFileDescriptor> moduleDescriptors) {
-      myNodePathsMap = getPathsMap(moduleDescriptors);
-      addDependencies(moduleDescriptors);
-    }
-
-    @Override
-    public Collection<Node> getNodes() {
-      return myNodePathsMap.values();
-    }
-
-    @Override
-    public Iterator<Node> getIn(Node node) {
-      return node.myDependencies.iterator();
-    }
-
-    @Nullable
-    public Node getNode(@NotNull String moduleName) {
-      return myNodePathsMap.get(moduleName);
-    }
-
-    private void addDependencies(List<ErlangFileDescriptor> moduleDescriptors) {
-      for (ErlangFileDescriptor descriptor : moduleDescriptors) {
-        Node node = myNodePathsMap.get(descriptor.myErlangModulePath);
-        node.myDependencies.addAll(getNodesByName(descriptor.myDependencies));
+  @NotNull
+  private static List<String> findAppFiles(@NotNull Collection<String> removedFiles) {
+    return ContainerUtil.filter(removedFiles, new Condition<String>() {
+      @Override
+      public boolean value(String filePath) {
+        return isAppConfigFileName(filePath);
       }
-    }
-
-    @NotNull
-    private List<Node> getNodesByName(@NotNull Collection<String> nodes) {
-      return ContainerUtil.mapNotNull(nodes, new Function<String, Node>() {
-        @Override
-        public Node fun(String dependencyName) {
-          return myNodePathsMap.get(dependencyName);
-        }
-      });
-    }
+    });
+  }
 
-    private static LinkedHashMap<String, Node> getPathsMap(List<ErlangFileDescriptor> moduleDescriptors) {
-      LinkedHashMap<String, Node> nodePathsMap = ContainerUtil.newLinkedHashMap();
-      for (ErlangFileDescriptor descriptor : moduleDescriptors) {
-        nodePathsMap.put(descriptor.myErlangModulePath, new Node(descriptor.myErlangModulePath));
+  @NotNull
+  private static List<String> findErlFiles(@NotNull Collection<String> removedFiles) {
+    return ContainerUtil.filter(removedFiles, new Condition<String>() {
+      @Override
+      public boolean value(String s) {
+        return s.endsWith(".erl");
```

Comment:
Extract a method for this. (You now have two 'ends with ".erl"' checks in your code)